lego_manager/lego_manager.py

```diff
@@ -15,53 +16,119 @@ class LegoManager(rpyc.Service):
     __slots__ = ("_allocations", "_bg_threads")
     ALIASES = ["LegoManager"]
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: str, **kwargs: str) -> None:
         super().__init__(*args, **kwargs)
-        self._allocations = dict()
-        self._bg_threads = dict()
+        self._allocations: Dict = dict()
+        self._bg_threads: Dict = dict()
+
+    def on_connect(self, conn: rpyc.Connection) -> None:
+        """Initializes thread for the incoming connection.
+
+        Args:
+            conn: An incoming connection.
+        """
 
-    def on_connect(self, conn):
         self._bg_threads[conn] = rpyc.BgServingThread(conn)
 
-    def on_disconnect(self, conn):
+    def on_disconnect(self, conn: rpyc.Connection) -> None:
+        """Stops the connection's thread.
+
+        Args:
+            conn: A disconnected connection.
+        """
+
         self._bg_threads.pop(conn).stop()
 
     @contextlib.contextmanager
-    def allocation(self, components):
-        self.allocate(components)
+    def _allocation(
+            self,
+            _components: List[Tuple[str, str]],
```

Comment:
Why it is named _components?

Reply:
Because of 'components' is a directory name that creates duplication.

Reply:
We can think about another name for the fixture / directory

Reply:
I don't like the `_components`.
Please change to a normal name